### DISCUSSION TITLE
Pretty URL for .html by default?

### DISCUSSION THREAD

**Discussion by miro-ux** [leokhoa/laragon#404] [General]
Hi,

I have the following in `httpd.config` to expose my projects to Apache:
```
Alias /projects c:/projects

<Directory c:/projects >
	Options Indexes MultiViews FollowSymLinks
	AllowOverride All
	Require all granted
</Directory>
```

I've noticed that if you have `localhost/projects/page.html`, it can be also accessed by `localhost/projects/page` with the exact same content as `page.html`. There is no folder `page`. There is no `.htaccess` anywhere.

Also, it doesn't work via `laragon/www/`... only under `c:/projects`.

Can you explain this behavior and how to disable it. (Working on something that makes this feature stand in the way.)

**Comment by Pen-y-Fan** [comment]
Are you overriding Laragon's config with your own? If so I would recommend you delete your config and leave Laragon auto-create the config for you.

## Explanation of how Laragon works

Laragon will automatically create pretty URLs for projects in Laragon's www directory, based on the folder name. e.g. c:\laragon\www\project

Create an **index.html** file.  E.g. with \<h1>Hello, world\</h1>

![index.html](https://user-images.githubusercontent.com/40126936/217910431-8c71b873-14ba-4c9f-a4da-d41897aa9c04.png)

Stop and start Laragon (you may need to do it twice 🤷) 

Laragon will detect the folder and autogenerate the C:\laragon\etc\apache2\sites-enabled\\**auto.project.test.conf**

Laragon will also update the C:\Windows\System32\drivers\etc\\**host** file with:

```text
127.0.0.1      project.test         #laragon magic! 
```

Open the project Laragon menu > www > project

The default browser will open, to the pretty URL of **project.test**. 

![Hello, world pretty url](https://user-images.githubusercontent.com/40126936/217910316-c8874212-3e38-471a-9853-54650cce83c3.png)

**Comment by miro-ux** [comment]
@Pen-y-Fan thanks for replying. However, that's not what I'm asking there. Please read the post more thoroughly.
I'm taking about `/page.html` getting served when going to just `/page`... It's different than the `example.test` virtualHost setup that Laragon offers out the box.

**Reply by Pen-y-Fan** [reply]
Example project:

![image](https://user-images.githubusercontent.com/40126936/217962154-fffc1316-a846-446e-bca4-3489609931a9.png)

I'm unable to replicate the problem:

|  URL | Apache result |
| :--- | :--- |
| project.test | Ok - index.html rendered |
| project.test/page | 404 |
| project.test/page.html | Ok - page.html rendered |
| localhost/project | Ok - index.html rendered |
| localhost/project/page | 404 |
| localhost/project/page.html | Ok  - page.html rendered |

It may be related to **MultiViews** see <https://httpd.apache.org/docs/2.4/content-negotiation.html>
